programs/vote/src/vote_state/mod.rs

```diff
@@ -383,17 +383,15 @@ fn check_update_vote_state_slots_are_valid(
             true
         } else if vote_state_update_index == vote_state_update_indexes_to_filter[filter_votes_index]
         {
-            filter_votes_index = filter_votes_index.checked_add(1).expect(
-                "`filter_votes_index` is bounded by `vote_state_update_indexes_to_filter.len()`",
-            );
+            filter_votes_index = filter_votes_index.checked_add(1).unwrap();
```

Comment:
think this can just be an unwrap since the array access above will fail too

Reply:
The `retain` can only run `vote_state_update.lockouts.len()` number of iterations, so it's bounded by that